### ISSUE TITLE
includes/error highlighting confused when building project with bazel

### ISSUE THREAD

**Issue by nevion** [bazelbuild/intellij#330]
I have a project that demonstrates a weird problem with the clion bazel plugin (no sharing, sorry).  Clion can know files, symbols in terms of searching via Navigation but when I'm inside source files (c++),  clion seems not to be aware of all the things contained in headers I have #included .  It knows about _some_, but not _all_, and it is very frequent not finding symbols.

Error messages such as:

- Can't resolve <symbol>
- Cannot resolve constructor 'thing'
- Can't resolve variable 'variable'

It will also ask me do I want to import (via alt+enter) "../../some/ugly/relative/path.h" ,even though I've already got #includes that work and compile with clion/bazel.   

I'm not quite sure I can generate a minimal example to trigger this behaviour, but I've observed it for months now.  Any ideas to help debug what's going wrong?

**Comment by brendandouglas** [comment]
The spurious resolve errors are likely an issue with CLion itself -- hard to say for sure without more details or steps to reproduce.

**Comment by nevion** [comment]
I can say that when I used cmake for the same project, there was no issues like this, and I think those ugly relative path inclusions are a tip into why / what's happening.

Is there any way I can debug to see what the plugin is piping into clion for indexing or a file I'm editing?

**Comment by jvoung** [comment]
Some helpers to debug:
* Add cidr.show.compiler.info=true to idea.properties
* Run "Show Compiler Info" with the problematic C++ file open in an editor (say foo.cc). Not sure which menu that action is in, but "Help > Find Action" should find it

That'll show a lot of compiler flags like the -std=lang_level, as well as the 'Header Search paths' and #defines. The plugin determines the flags and search paths from the BUILD files and Crosstool.

Check that the Header Search paths are sufficient to resolve the #include 'whatever/path.h' from foo.cc, without resorting to the "../../some/ugly/relative/whatever/path.h".

Otherwise, it's a bit hard to debug without further details.

**Comment by nevion** [comment]
with the patches in #329 applied , I opened a few files and I have fairly blank pages for the compiler info, lightly edited example for a cpp file:

```
Compiler info for 'source.cpp' in configuration '//module:library' for language CPP

Compiler kind: Clang path: /tmp/blaze_compiler1.sh

Compiler switches:

null

Compiler info:

Defines:

Features:


PCHs:


Header Search paths:
```

**Comment by jvoung** [comment]
If the cpp file is marked "unsynced" then the plugin didn't get compiler info about it from bazel (flags, defines, header search roots).

Is the cpp file's bazel target not reachable from your ".bazelproject" file's targets? (see https://ij.bazel.build/docs/project-views.html#targets)

Or are the cc_tests marked "manual", and you used wildcards in the targets section?
"Targets with tags=["manual"] will not be included in wildcard target patterns (..., :*, :all, etc)." (see https://docs.bazel.build/versions/master/user-manual.html#target-patterns)

Why did you need the patches from #329? Do you mean it is supposed to contain C++ source, but the file extension is ".cu" or ".cuh"? What happens if you change the file extension? There might be additional file extension checks in CLion itself, not just the bazel plugin.

**Comment by nevion** [comment]
I did have targets restricted to just one entry that I usually work on as when I hit build, I'd like to just build that (couldn't find any other way to choose what targets to build with say Control+F9, this could use a checkbox based UI selector, sort of like qt-creators approach).  I didn't have manual selections on the tests.  So I changed the targets list back to ```//...:all``` and that fixed the unsynced files issue (yay!)  Followed up with a project sync, just to see if that fixed anything else.

To clarify the unsynced issue was only one of several issues that I've been having, unfortunately the compiler info exactly the same (essentially blank) for every cpp file I try - test, binary, or library.

As for the reference to 329, I figured it may have resulted in a broken code model having a mixed filetype project - what I'm directly reporting on is pure cpp files, though.   Worth mentioning the editor now seems to understand substantially more about cuda files (there were syntax highlighting changes), but I still have a ton of  "Can't find/resolve" issues, I wonder if some stray cuda syntax can result in a broken code model?  But to the extent of can't find / resolve errors and the blank compiler info?
